### PR TITLE
Docker updates to build and run 

### PR DESCRIPTION
**Pull request by franklywatson**
This is the basic starting point for more docker automation and is enough to get someone started with just some EVN variables and the make commands 

## Release Notes

- **New Features**
	- Introduced a new `flow.json` configuration file for managing network and account settings.
	- Enhanced Docker build process with new architecture and versioning arguments.
	- Updated README.md to clarify EVM Gateway functionalities and usage instructions.
	- Implemented functionality for managing and providing access to blockchain block snapshots within the replayer package.
	- Added transaction execution logic within a blockchain framework with robust error handling.
	- Introduced `RegisterStorage` for managing register state changes temporarily.
	- Added `RegisterDelta` functionality for managing register state changes without immediate persistence.
	- Added new tests for debugging transaction and call traces using Web3.js, including tests for retrieving transaction traces and validating contract interactions.
	- Added new systemd service file for managing the EVM Gateway running with Docker.
	- Introduced a new environment configuration file for runtime settings of the EVM Gateway.
	- Expanded test suite for Ethereum interactions, including new scenarios for verifying contract methods and state changes.

- **Bug Fixes**
	- Removed outdated build arguments in the Dockerfile and workflow.
	- Addressed issues with the handling of missing receipts and improved error handling throughout the storage methods.

- **Documentation**
	- Improved clarity and detail in the README.md, including new sections on key concepts and Docker usage.

- **Chores**
	- Updated `.gitignore` to exclude specific directories and files from version control.
	- Removed the `dev/run.sh` script as part of streamlining the setup process.
	- Updated test configurations and improved the testing framework for various functionalities.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,14 +91,22 @@ need funded accounts for interacting with EVM._
...
 **With Docker**
 
-Run the following commands:
+Set the following ENV variables 
 
 ```bash
-cd dev
+FLOW_NETWORK_ID=[flow-testnet | flow-mainnet]
+ACCESS_NODE_GRPC_HOST=
+INIT_CADENCE_HEIGHT=211176670
```
Comment (m-Peter) [inline README.md:99]: This is correct for `testnet`, but I think it's a different value for `mainnet`. Maybe we should point this out too.

Comment (franklywatson) [reply README.md:99]: Yeah, I maybe should have put the PR as draft :)

Comment (m-Peter) [reply README.md:99]: No worries :sweat_smile:

**Comment by franklywatson** [comment]
PR now has updates to address various build and architecture issues and an improved README.md. 

The README now strictly covers only the development and local build/test related concerns, with all other documentation on gateway operations being mastered in the [Flow public docs](https://github.com/onflow/docs/pull/1035)

File: Makefile [hunk 2]
```diff
@@ -44,37 +103,104 @@ ci: check-tidy test e2e-test
...
+	docker run -p 8545:8545 --add-host=host.docker.internal:host-gateway "$(CONTAINER_REGISTRY)/evm-gateway:$(COMMIT)" $(CMD_ARGS)
+
+
+# Build docker image for release
+.PHONY: docker-build
+docker-build:
+ifdef GOARCH
+	$(eval ARCH=$(GOARCH))
+endif
+	docker build --build-arg VERSION="$(VERSION)" --build-arg ARCH=$(ARCH)  -f Dockerfile -t "$(CONTAINER_REGISTRY)/evm-gateway:$(IMAGE_TAG)" \
```
Comment (franklywatson) [inline Makefile:150]: @vishalchangrani - let me know what should be done for tags - i copied what we do for other Flow nodes, but not sure that is what you need

**Review by j1010001 — APPROVED** [review]

**Pull request merged**